File: src/Domain/Entities/Subject.cs
namespace Domain.Entities;

public class Subject
{
    public Guid SubjectId { get; set; }
    public string Name { get; set; } = null!;
    public string Description { get; set; } = null!;
    public Guid LecturerId { get; set; }
    public Guid GroupId { get; set; }
    public Lecturer Lecturer { get; set; } = null!;
    public Group Group { get; set; } = null!;
    public List<Task> Tasks { get; set; } = null!;
}

File: src/Web/src/app/core/models/subject.ts
import {LecturerTask, StudentTask, StudentTasks} from "./task";
import {Group} from "./group";

export interface LecturerSubjectsResponse {
  subjects: LecturerSubject[];
}

export interface LecturerSubject {
  subjectId: string;
  name: string;
  description: string;
  group: Group;
  tasks: LecturerTask[];
}

export interface StudentSubjectsResponse {
  subjects: StudentSubject[];
}

export interface StudentSubject {
  subjectId: string,
  name: string,
  description: string,
  lecturerName: string,
  tasks: StudentTask[],
  totalGrade: number,
  averageGrade: number,
}

export interface SubjectGrades {
  subjectId: string,
  subjectName: string,
  groupName: string,
  studentTasks: StudentTasks[]
}

export interface CreateSubjectRequest {
  name: string;
  description: string;
  group: string;
}
